lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -594,6 +602,11 @@ def write(chunk)
             unless record_json.nil?
               record = record_json
               is_json = true
+              # Extract LogEntry fields if they are present in the nested JSON.
+              # This will take precedence of the root level values.
+              fq_trace_id ||= record.delete(@trace_key)
```

Comment:
This is wrong too, since the outer fields would override the nested fields, while it should be the other way around.

Reply:
Switched the order in https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/247/commits/94a2ff67c58ec2d2995f7b032858f6c36b08ae8d.

I would propose this way instead of mocking exactly what we do with `time` and `severity` fields because:
For `time` and `severity` fields, we are not deleting them from the record in the end, so it makes sense to put them back into the record. For `trace`, `spanId` and `insertId`, we eventually delete them. So just storing the value in local variables should be good enough.

WDYT?

Reply:
Updated. PTAL